app/src/main/java/com/parking_system_kotlin/activities/ReservationActivity.kt

```diff
@@ -0,0 +1,47 @@
+package com.parking_system_kotlin.activities
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.parking_system_kotlin.listeners.ListenerDateTime
+import com.parking_system_kotlin.mvp.contracts.ParkingReservationContract
+import com.parking_system_kotlin.mvp.presenter.ParkingReservationPresenter
+import com.parking_system_kotlin.mvp.view.ParkingReservationView
+import com.parkingsystemkotlin.databinding.ActivityReservationBinding
+import java.util.*
+
+class ReservationActivity : AppCompatActivity(), ListenerDateTime {
+
+    private lateinit var binding: ActivityReservationBinding
+    private lateinit var presenter: ParkingReservationContract.ParkingReservationPresenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityReservationBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        presenter = ParkingReservationPresenter(ParkingReservationView(this, binding))
+
+        setListeners()
+    }
+
+    fun setListeners() {
+        binding.editTextReservationActivityEntry.setOnClickListener { presenter.showDatePicker(this, true) }
+        binding.editTextReservationActivityExit.setOnClickListener { presenter.showDatePicker(this, false) }
+        binding.buttonReservationActivitySave.setOnClickListener { presenter.saveReservation() }
```

Comment:
You can use and apply here with binding. Instead of call binding three times you can change it with:
binding.apply {
    editTextReservationActivityEntry...
    editTextReservationActivityExit...
    buttonReservationActivitySave...
}

Reply:
done

Reply:
in this case you shuold use with(binding){ ... } instead of apply because you're not defining some behavior of the binding

Reply:
ohhh okey! changed!